src/Lecture2.hs

```diff
@@ -175,64 +273,71 @@ challenging. However, you still may find some of them easier than some of the
 previous ones. Difficulty is a relative concept.
 -}
 
-{- | Write a function that takes a list of numbers and returns 'True'
-if all the numbers are in the increasing order (i.e. the list is
-sorted).
-
->>> isIncreasing [3, 1, 2]
-False
->>> isIncreasing [1 .. 10]
-True
--}
+-- | Write a function that takes a list of numbers and returns 'True'
+-- if all the numbers are in the increasing order (i.e. the list is
+-- sorted).
+--
+-- >>> isIncreasing [3, 1, 2]
+-- False
+-- >>> isIncreasing [1 .. 10]
+-- True
 isIncreasing :: [Int] -> Bool
-isIncreasing = error "TODO"
-
-{- | Implement a function that takes two lists, sorted in the
-increasing order, and merges them into new list, also sorted in the
-increasing order.
-
-The lists are guaranteed to be given sorted, so you don't need to
-verify that.
-
->>> merge [1, 2, 4] [3, 7]
-[1,2,3,4,7]
--}
+isIncreasing lst = case lst of
+  [] -> True
+  [_] -> True
+  (x : xs) -> (x < head xs) && isIncreasing xs
+
+-- | Implement a function that takes two lists, sorted in the
+-- increasing order, and merges them into new list, also sorted in the
+-- increasing order.
+--
+-- The lists are guaranteed to be given sorted, so you don't need to
+-- verify that.
+--
+-- >>> merge [1, 2, 4] [3, 7]
+-- [1,2,3,4,7]
 merge :: [Int] -> [Int] -> [Int]
-merge = error "TODO"
-
-{- | Implement the "Merge Sort" algorithm in Haskell. The @mergeSort@
-function takes a list of numbers and returns a new list containing the
-same numbers but in the increasing order.
-
-The algorithm of merge sort is the following:
-
-  1. If the given list has less than 2 elements, it's already sorted.
-  2. Otherwise, split list into two lists of the same size.
-  3. Sort each of two lists recursively.
-  4. Merge two resulting sorted lists to get a new sorted list.
-
->>> mergeSort [3, 1, 2]
-[1,2,3]
--}
+merge [] b = b
+merge a [] = a
+merge (xa : xsa) (xb : xsb)
+  | xa < xb = xa : merge xsa (xb : xsb)
+  | otherwise = xb : merge (xa : xsa) xsb
```

Comment:
Absolutely correct solution! 💎 
You can make it more performant by checking if both heads of the given lists are equal. In that case, you're able to reduce the size of each list by 1 on each recursive call instead of reducing the size of only a single list ✂️